### PR TITLE
[CALCITE-5553] noFlatteningForInput handles single field structs

### PR DESCRIPTION
**Pull request by apilloud**

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java [hunk 1]
```diff
@@ -291,13 +297,10 @@ protected MockCatalogReaderSimple(RelDataTypeFactory typeFactory,
     registerTable(suppliersTable);
 
     // Register "EMP_20" and "EMPNULLABLES_20 views.
-    // Same columns as "EMP" amd "EMPNULLABLES",
-    // but "DEPTNO" not visible and set to 20 by default
-    // and "SAL" is visible but must be greater than 1000,
-    // which is the equivalent of:
+    // Same columns as "EMP" amd "EMPNULLABLES", but "DEPTNO" not visible and set to 20 by default
+    // and "SAL" is visible but must be greater than 1000, which is the equivalent of:
```
Comment (libenchao) [inline testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java:301]: Is it necessary to do this comment reformatting?

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 2]
```diff
@@ -3709,6 +3709,12 @@ void checkCorrelatedMapSubQuery(boolean expand) {
     sql(sql).ok();
   }
 
+  @Test void testNestedStructFieldAccessWhere() {
```
Comment (libenchao) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java:3712]: The test name does not explicitly say that it must be a single field struct.

**Review by libenchao — COMMENTED** [review]
@apilloud Thanks for your PR, the fix looks good to me generally, I've only two minor comments about the tests.

File: testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java (hunk 1 above)
Comment (apilloud) [reply testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java:301]: Without the comment reformatting I get the following test failure:
```
/home/runner/work/calcite/calcite/testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java:79:3: Method length is 376 lines (max allowed is 370). [MethodLength]
```
This seemed like the simplest change, but I'm happy to do something else instead.

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java (hunk 2 above)
Comment (apilloud) [reply core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java]: Renamed to `testNestedStructSingleFieldAccessWhere`

File: testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java (hunk 1 above)
Comment (libenchao) [reply testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java:301]: I see, then we'd better split this method into smaller ones in a separate issue, and I'm ok for this workaround in this one.

**Review by libenchao — COMMENTED** [review]
@apilloud Thanks for the updating, the changes looks good now. There is only one thing left, the tests fails due to the orderness in `SqlToRelConverterTest.xml`.

File: testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java (hunk 1 above)
Comment (libenchao) [reply testkit/src/main/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java:301]: I've logged CALCITE-5574 for this.

**Review by libenchao — APPROVED** [review]
+1, will merge after the main branch is open.

**Pull request closed without merging**